### PR TITLE
Logs artifact finding, breaking, effects, anomaly batteries use and harvesting

### LINKED ISSUE #6649: Artifact activation is not logged.
It'd be very useful to know if someone say, flooded the shuttle with plasma with an artifact.

### PR DESCRIPTION
**Pull request by jknpj**
log
Adds artifact investigation log with findings, breaking, effect triggering, button pressing, spawning and effect harvesting.

Closes #6649

:cl:

 * bugfix: Gas/Temp triggered secondary artifact effects can be toggled off now

### CODE REVIEW

File: code/modules/mining/mine_turfs.dm [hunk 1]
```diff
@@ -303,6 +303,7 @@ turf/unsimulated/mineral/ChangeTurf(var/turf/N, var/tell_universe=1, var/force_l
 						B = getFromPool(/obj/structure/boulder, src)
 						if(artifact_find)
 							B.artifact_find = artifact_find
+							B.investigation_log(I_ARTIFACT, "|| [B.artifact_find.artifact_find_type] || found by [user.name]([user.ckey]).")
```
Comment (ComicIronic) [inline code/modules/mining/mine_turfs.dm]: Use the proc for formatting ckey and mob name properly.

**Review by ComicIronic — CHANGES_REQUESTED** [review]

File: code/modules/mining/mine_turfs.dm (hunk 1 above)
Comment (jknpj) [reply code/modules/mining/mine_turfs.dm]: It is using user from `/turf/unsimulated/mineral/attackby(obj/item/weapon/W as obj, mob/user as mob)`
user
This log is from before adding that second ||

Comment (ComicIronic) [reply code/modules/mining/mine_turfs.dm]: Yes, but `key_name(user)` is the correct way to do it, instead of manual formatting.

Comment (jknpj) [reply code/modules/mining/mine_turfs.dm]: The more you know

**Review by ComicIronic — APPROVED** [review]

**Comment by ComicIronic** [comment]
Let's see if this breaks something.

**Pull request merged**